### ISSUE TITLE
New Component: Custom Select

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5226]
## What is your proposal?

**Create a new clay component, Custom Select, or Custom Combobox Select.**

![image](https://user-images.githubusercontent.com/7963804/205102104-798c33e1-34f3-443f-b03d-286858287e03.png)

Context:
From some accessibility issues in DXP like this: https://issues.liferay.com/browse/LPS-167535 we decided to init the debate here: https://github.com/liferay/clay/issues/5181
After the debate about change dropdown to make it configurable as a select we finally decided to create a new component, we created a design task:

Design task:
https://issues.liferay.com/browse/DESIGN-4801

Design final documentation:
https://liferay.atlassian.net/wiki/spaces/ENGLEXICON/pages/2178057030/Custom+Select

@emiliano-cicero's design phase defines the component quite detailed, here we can highlight some things:

- Now, the component visually looks like a select
- We recommend to provide native one when possible in mobile, in order to make the component much more usable and accessible, here a quick explanation:
  - https://liferay.atlassian.net/wiki/spaces/ENGLEXICON/pages/2179170308/Custom+VS+Native
  - If make hybrid component is not possible at the beginning, we can make the same view for mobile and  study how to be visually closer to native select view

## Technical side:
We need to apply roles and markup from https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-select-only.html
With all improvements we could make: https://github.com/liferay/clay/issues/5181#issuecomment-1316005047
Of course if we can provide in someway (ScreenReader, markup validation, etc). the suggested solution is better.

The componen needs to be defined following 

## Why would adopting this proposal be beneficial?

It will allow us to fix tons of issues in DXP, and will allow designers to provide better solution for 

## What are the alternatives to this proposal?

We already discussed if make this as a part of the dropdown menu could be good idea:
https://github.com/liferay/clay/issues/5181

**Comment by pat270** [comment]
@marcoscv-work The Custom Select default style is slightly different from the regular select style. White background instead of gray, is this intentional?

![regular-select](https://user-images.githubusercontent.com/788266/205185580-04600a93-6b9d-4fa7-83da-db1aea126d14.jpg)

**Comment by emiliano-cicero** [comment]
hey @pat270, yes that's intentional because in that example I'm using a button (with a custom fixed size -- do you think we need a fixed minimum?). It should be possible to trigger the custom select panel from a button or an input for the different use cases that we have. For example, the management toolbar "view" button will probably use the custom select, but also the different form inputs as always. 

And to make it clear: we're keeping the same input style. 

(Thanks for asking! I'll add all these details to the docs soon).

**Comment by pat270** [comment]
@emiliano-cicero I think the fixed minimum should be 0. The same as the regular select element. I'm going to have an issue with the name Custom Select though. Bootstrap has a `custom-select` component and this will conflict with that.

I'd like to piggy back off of the select styles. Clay has `form-control-select` that can be used on any element. We can create variants for it like `form-control-secondary`.
